hub/api/dataset.py

```diff
@@ -101,6 +110,20 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def flush(self):
+        """Necessary operation after writes if a cache is being used. Clears the cache and sends the data to underlying storage"""
+        self.storage.flush()
+
+    def clear_cache(self):
+        """Flushes the contents of the cache and deletes contents of all the layers of it.
+        This doesn't clear data from the actual storage.
+        This is useful if you have multiple dataset with memory caches open, taking up too much RAM.
```

Comment:
We need to watch out for using internal language in our user-facing documentation. Things like "storage cache" can get redundant or hard-to-read.